### PR TITLE
check token before checking permissions

### PR DESCRIPTION
**Pull request by JarekW**
if the user enters a "`404`" page that's behind firewall and also the user is authorized, calling `->isGranted(...)` will cause:

> AuthenticationCredentialsNotFoundException
> The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL.

404 is then changed with 500 error. This is due that url's are resolved before firewall starts

### CODE REVIEW

**Comment by JarekW** [comment]
@fsi-open

File: EventListener/EnforcePasswordChangeListener.php [hunk 1]
```diff
@@ -95,11 +96,16 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
+        $token = $this->tokenStorage->getToken();
+        if (!$token instanceof TokenInterface) {
```
Comment (szymach) [inline EventListener/EnforcePasswordChangeListener.php]: isn't `if (!$token)` enough?

**Comment by szymach** [comment]
:+1:

**Comment by chives** [comment]
:+1:

**Comment by norzechowicz** [comment]
just like other PRs, this one also should be pointed to master branch

**Comment by JarekW** [comment]
so this will be after ancient #32

**Comment by norzechowicz** [comment]
it would be good to work a bit on #32 before merging anything else

**Comment by chives** [comment]
@JarekW please rebase this PR over current master branch

**Pull request merged**